assets/app/scripts/directives/logViewer.js

```diff
@@ -224,6 +228,35 @@ angular.module('openshiftConsole')
               goChromeless: logLinks.chromelessLink,
               restartLogs: streamLogs
             });
+
+            APIDiscovery
+              .getLoggingURL()
+              .then(function(url) {
+                // TODO: update lodash for _.get, would rather do this:
+                //var projectName = _.get($scope.context, 'project', 'metadata', 'name');
+                var projectName = $scope.context &&
+                                  $scope.context.project &&
+                                  $scope.context.project.metadata &&
+                                  $scope.context.project.metadata.name;
+
+                var hasComponents = projectName && $scope.name && url;
```

Comment:
I suggest we use `$watchGroup()` to observe changes to `context.project.metadata.name` and `name` and update the Kibana URL. Without the watch, I'm not sure if we should even bother with the null check since nothing in the log viewer directive will work when those aren't defined. We watch `name` above, so we should be consistent.

If we add `$watchGroup`, keep the `getLoggingURL()` call outside the callback so we don't run it more than once.